clients/java/signalr/src/main/java/com/microsoft/aspnet/signalr/WebSocketTransport.java

```diff
@@ -59,11 +60,29 @@ public void start() throws Exception {
         logger.log(LogLevel.Information, "WebSocket transport connected to: %s", webSocketClient.getURI());
     }
 
+    @Override
+    public CompletableFuture startAsync() {
+        return CompletableFuture.runAsync(() -> {
+            webSocketClient = createWebSocket(headers);
+            try {
+                webSocketClient.connectBlocking();
+            } catch (InterruptedException e) {
+                e.printStackTrace();
```

Comment:
We rethrow or something

Reply:
Was having some trouble with that actually. This actually fails one of the tests because the exception gets smothered. But I wasn't able to re-throw. Figuring that out now

Reply:
Figured out that we have to throw Runtime Exceptions here